lib/tests/streamlit/dataframe_util_test.py

```diff
@@ -607,12 +615,249 @@ class StrOpt(str, enum.Enum):
+            ),
+            # Tuple:
+            ((), []),
+            (
+                ("st.number_input", "st.text_area", "st.text_input"),
+                ["st.number_input", "st.text_area", "st.text_input"],
+            ),
+            # Dict:
+            ({}, []),
+            (
+                {
+                    "st.number_input": "number",
+                    "st.text_area": "text",
+                    "st.text_input": "text",
+                },
+                ["st.number_input", "st.text_area", "st.text_input"],
+            ),
+            # Dict keys:
+            (
+                {
+                    "st.number_input": "number",
+                    "st.text_area": "text",
+                    "st.text_input": "text",
+                }.keys(),
+                ["st.number_input", "st.text_area", "st.text_input"],
+            ),
+            # Dict values:
+            (
+                {
+                    "st.number_input": "number",
+                    "st.text_area": "text",
+                    "st.text_input": "text",
+                }.values(),
+                ["number", "text", "text"],
+            ),
+            # OrderedDict:
+            (
+                OrderedDict(
+                    [
+                        ("st.number_input", "number"),
+                        ("st.text_area", "text"),
+                        ("st.text_input", "text"),
+                    ]
+                ),
+                ["st.number_input", "st.text_area", "st.text_input"],
+            ),
+            # Enum:
+            (
+                TestEnum,
+                [TestEnum.NUMBER_INPUT, TestEnum.TEXT_AREA, TestEnum.TEXT_INPUT],
+            ),
+            (StrTestEnum, ["st.number_input", "st.text_area", "st.text_input"]),
+            # Generator:
+            (data_generator(), ["st.number_input", "st.text_area", "st.text_input"]),
+            # Enumerate:
+            (
+                enumerate(["st.number_input", "st.text_area", "st.text_input"]),
+                [0, 1, 2],
+            ),
+            # Range:
+            (range(3), [0, 1, 2]),
+            # Pandas Dataframe:
+            (
+                pd.DataFrame(),
+                [],
+            ),
+            (
+                pd.DataFrame(
+                    columns=["name", "type"], index=pd.RangeIndex(start=0, step=1)
+                ),
+                [],
+            ),
+            (
+                pd.DataFrame(["st.number_input", "st.text_area", "st.text_input"]),
+                ["st.number_input", "st.text_area", "st.text_input"],
+            ),
+            (
+                pd.DataFrame(
+                    {
+                        "widgets": ["st.number_input", "st.text_area", "st.text_input"],
+                        "types": ["number", "text", "text"],
+                    }
+                ),
+                ["st.number_input", "st.text_area", "st.text_input"],
+            ),
```

Comment:
nit: might be worth adding a quick explanatory comment about why the expected result is what it is. I'm guessing it's because the `widgets` column is treated as the index column, but this wasn't entirely clear to me when skimming through the cases

Reply:
Added a comment. The current logic is that the first column (in a dataframe-like structure) is always selected as the sequence.